### PR TITLE
[Storage] error overhaul

### PR DESCRIPTION
**Pull request by cindyxkuang**
Fixes # .

### Motivation
continuing from [https://github.com/coinbase/rosetta-sdk-go/pull/134](https://github.com/coinbase/rosetta-sdk-go/pull/134)

### Solution
added errors.go file in `storage` package and unit test

### CODE REVIEW

File: keys/errors.go [hunk 1]
```diff
@@ -25,6 +25,10 @@ var (
 	ErrPrivKeyUndecodable   = errors.New("could not decode privkey")
 	ErrPrivKeyLengthInvalid = errors.New("invalid privkey length")
 
+	ErrKeyGenSecp256k1Failed    = errors.New("keygen: new private key")
```
Comment (qiwu7) [inline keys/errors.go:28]: nit: error message can be rephrased to be more relevant to the actual error
```suggestion
	ErrKeyGenSecp256k1Failed    = errors.New("keygen: error generating keypair for secp256k1 curve type")
```

File: storage/errors.go [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+	utils "github.com/coinbase/rosetta-sdk-go/errors"
+)
+
+// Named error types for Storage errors
+var (
+	////////////////////
+	/* BADGER STORAGE */
+	////////////////////
+
+	ErrDatabaseOpenFailed         = errors.New("could not open database")
```
Comment (qiwu7) [inline storage/errors.go:29]: nit: to be consistent with other error messages. Maybe it's better to update to the following
```suggestion
	ErrDatabaseOpenFailed         = errors.New("unable to open database")
```

File: storage/errors.go [hunk 3]
```diff
@@ -0,0 +1,369 @@
...
+	ErrCompressorLoadFailed       = errors.New("could not load compressor")
```
Comment (qiwu7) [inline storage/errors.go:30]: same nit here
```suggestion
	ErrCompressorLoadFailed       = errors.New("unable to load compressor")
```

File: storage/errors.go [hunk 4]
```diff
@@ -0,0 +1,369 @@
...
+	ErrStoreKeyFailed           = errors.New("unable to store key")
+	ErrCommitKeyFailed          = errors.New("unable to commit new key to db")
+	ErrAddrGetFailed            = errors.New("unable to get address")
+	ErrAddrNotFound             = errors.New("address not found")
+	ErrParseSavedKeyFailed      = errors.New("unable to parse saved key")
+	ErrKeyScanFailed            = errors.New("database scan for keys failed")
+	ErrParseKeyPairFailed       = errors.New("unable to parse key pair")
+	ErrKeyGetFailed             = errors.New("could not get key")
+	ErrSignerCreateFailed       = errors.New("unable to create signer")
+	ErrDetermineSigTypeFailed   = errors.New("cannot determine signature type for payload")
```
Comment (qiwu7) [inline storage/errors.go:280]: similar nit here
```suggestion
	ErrDetermineSigTypeFailed   = errors.New("unable to determine signature type for payload")
```

File: keys/errors.go (hunk 1 above)
Comment (patrick-ogrady) [reply keys/errors.go:28]: Agree with @qiwu7's nit here!

Comment (cindyxkuang) [reply keys/errors.go]: 👍

File: storage/badger_storage.go [hunk 5]
```diff
@@ -565,19 +561,19 @@ func BadgerTrain(
...
+				return ErrMaxEntries
 			}
 
 			return nil
 		},
 		true,
 	)
-	if err != nil && !errors.Is(err, errMaxEntries) {
+	if err != nil && !errors.Is(err, ErrMaxEntries) {
 		return -1, -1, fmt.Errorf("%w: unable to scan for %s", err, namespace)
```
Comment (qiwu7) [inline storage/badger_storage.go:572]: nit: do we need to extract this error message into it's own const as well?

Comment (cindyxkuang) [reply storage/badger_storage.go]: good catch, I had taken it out while I fixed breaking tests but can add it back in now!

**Review by patrick-ogrady — APPROVED** [review]
LGTM!

**Pull request merged**